File: codeforces/1622C/src/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    BufferedReader in;
    PrintWriter out;

    Main() {
        in = new BufferedReader(new InputStreamReader(System.in));
        out = new PrintWriter(System.out);
    }

    void close() throws IOException {
        in.close();
        out.flush();
        out.close();
    }

    public static void main(String[] args) throws IOException {
        Main m = new Main();
        m.solve();
        m.close();
    }
    
    void solve() throws IOException {
        int t = Integer.parseInt(in.readLine());
        while (t > 0) {
            t--;
            StringTokenizer st = new StringTokenizer(in.readLine());
            int n = Integer.parseInt(st.nextToken());
            long k = Long.parseLong(st.nextToken());
            st = new StringTokenizer(in.readLine());
            int[] a = new int[n];
            long s = 0;
            for (int i = 0; i < n; i++) {
                a[i] = Integer.parseInt(st.nextToken());
                s += a[i];
            }
            Arrays.sort(a);
            long ans;
            if (s <= k) {
                ans = 0;
            } else {
                ans = Integer.MAX_VALUE;
                for (int set = 0; set <= n - 1; set++) {
                    if (set >= ans) {
                        break;
                    }
                    long ns = s + (long) a[0] * set;
                    long decrease = 0;
                    if (ns > k) {
                        long d = ns - k;
                        decrease = (d + set) / (set + 1);
                    }
                    long step = set + decrease;
                    ans = Math.min(ans, step);
                    s -= a[n - 1 - set];
                }
            }
            out.append(String.format("%d\n", ans));
        }
    }

}

File: codeforces/1622C/main.cpp
#include <algorithm>
#include <cassert>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main()
{
#ifndef ONLINE_JUDGE
    freopen("1.in", "r", stdin);
#endif
    int t;
    cin >> t;
    while (t > 0) {
        t--;
        int n;
        long long k;
        cin >> n >> k;
        vector<int> a(n);
        long long s = 0;
        for (int i = 0; i < n; i++) {
            cin >> a[i];
            s += a[i];
        }
        sort(a.begin(), a.end());
        long long ans;
        if (s <= k) {
            ans = 0;
        } else {
            ans = 1 << 30;
            for (int set = 0; set <= n - 1; set++) {
                if (set >= ans) {
                    break;
                }
                long long ns = s + (long long)a[0] * set;
                long long decrease = 0;
                if (ns > k) {
                    decrease = (ns - k + set) / (set + 1);
                }
                long long step = decrease + set;
                ans = min(ans, step);
                s -= a[n - 1 - set];
            }
        }
        cout << ans << endl;
    }
    return 0;
}